test/cppsuite/test_harness/workload/database_model.h

```diff
@@ -49,112 +55,188 @@ struct value_t {
 };
 
 /* A collection is made of mapped Key objects. */
-struct collection_t {
-    std::map<key_value_t, key_t> keys;
-    std::map<key_value_t, value_t> values;
+class collection {
+    public:
+    collection(const uint64_t id, const uint64_t key_count, const std::string name)
+        : id(id), _key_count(key_count), name(name)
+    {
+    }
+
+    int
+    get_new_key()
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return _key_count++;
+    }
+
+    std::map<key_value_t, key_t>
+    get_keys(const uint64_t &id)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return (keys);
+    }
+
+    value_t
+    get_record(const uint64_t &id, const char *key)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        return (values.at(key));
+    }
+
+    void
+    insert_record(const uint64_t &id, const char *key, const char *value)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        keys[key].exists = true;
+        value_t v;
+        v.value = key_value_t(value);
+        values.emplace(key_value_t(key), v);
+    }
+
+    void
+    update_record(const uint64_t &id, const char *key, const char *value)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        values.at(key).value = key_value_t(value);
+    }
+
+    void
+    delete_record(const uint64_t &id, const char *key)
+    {
+        std::lock_guard<std::mutex> lg(_mtx);
+        keys.at(key).exists = false;
+    }
+
+    std::map<key_value_t, key_t> keys = {};
```

Comment:
Should we just have one map ?
std::map<key_value_t, key_value_t>

Reply:
Seems that way to me. Also, we don't need to have `= {}`. The default constructor will do the right thing.